var/spack/repos/builtin/packages/py-prometheus-client/package.py

```diff
@@ -11,14 +11,18 @@ class PyPrometheusClient(PythonPackage):
 
     pypi = "prometheus_client/prometheus_client-0.7.1.tar.gz"
 
+    version('0.12.0', sha256='1b12ba48cee33b9b0b9de64a1047cbd3c5f2d0ab6ebcead7ddda613a750ec3c5')
     version('0.7.1', sha256='71cd24a2b3eb335cb800c7159f423df1bd4dcd5171b234be15e3f31ec9f622da')
     version('0.7.0', sha256='ee0c90350595e4a9f36591f291e6f9933246ea67d7cd7d1d6139a9781b14eaae')
     version('0.5.0', sha256='e8c11ff5ca53de6c3d91e1510500611cafd1d247a937ec6c588a0a7cc3bef93c')
 
     variant('twisted', default=False, description='Expose metrics as a twisted resource')
 
     depends_on('py-setuptools', type='build')
-    depends_on('py-twisted', type=('build', 'run'), when='+twisted')
+    # Notice: prometheus_client/twisted/_exposition.py imports 'twisted.web.wsgi'
+    # which was not ported to Python 3 until twisted 16.0.0
+    depends_on('py-twisted@16:', type=('build', 'run'), when='@0.12.0: +twisted ^python@3:')
```

Comment:
This package does not require Python 3, so I see no reason to require twisted 16+. If you want to make this more explicit, I would add `depends_on('python@:2', when='@:15', type=('build', 'run'))` to the `py-twisted` package instead.

Reply:
@adamjstewart sorry, I don't understand your comment. Yes, there is a mistake here - the default dependency on py-twisted should stay, the restriction is only for python 3.

Reply:
Let's say we have 10 packages that depend on `py-twisted`. We could add `depends_on('py-twisted@16:', when='^python@3:')` to all 10 packages (and hope that we don't miss any), or we could instead add `depends_on('python@:2', when='@:15')` to the `py-twisted` package only.

Reply:
Ah. No, twisted 16 supports both py2 and py3, but the wsgi module was not ported to python3 until 16.0.0